File: add/codesnippet/VisualBasic/p-system.web.ui.datasour_3_1.vb
   ' The CsvDataSourceView does not currently
   ' permit deletion. You can modify or extend
   ' this sample to do so.
   Public Overrides ReadOnly Property CanDelete() As Boolean
      Get
         Return False
      End Get
   End Property

   Protected Overrides Function ExecuteDelete(keys As IDictionary, values As IDictionary) As Integer
      Throw New NotSupportedException()
   End Function 'ExecuteDelete

File: add/codesnippet/CSharp/p-system.web.ui.datasour_3_1.cs
        // The CsvDataSourceView does not currently
        // permit deletion. You can modify or extend
        // this sample to do so.
        public override bool CanDelete {
            get {
                return false;
            }
        }
        protected override int ExecuteDelete(IDictionary keys, IDictionary values)
        {
            throw new NotSupportedException();
        }